### DISCUSSION TITLE
Quest - Controller - Collision

### DISCUSSION THREAD

**Discussion by Efr3D** [microsoft/MixedRealityToolkit-Unity#9978] [Q&A]
Hi ,

When I use the controllers (Ques),   how can I do to have the same behavior with NearInteractionTouchableVolume?
In the hand Tracking Mode, I received OnTouchStarted....
But, when I use the controller (Quest), I never received OnTriggerEnter, or OnCollisionEnter.
I use a collider on my object. 
I have the feeling of having phantom hands, when I use the controllers.

What would be your advice to detect the collision of my controllers (Quest) with my objects.

Thanks

**Comment by Efr3D** [comment]
### Additional information

* MRTK 2.7.0
* Oculus XR Plugin 1.9.1
* Unity 3D 2019.4.27f1
* Oculus Quest 2 (SDK 29)

**Reply by RogPodge** [reply]
Currently near interactions are not supported with controllers on MRTK. You can add near pointers (the poke and grab pointers), by adjusting the pointer settings in the Input section of the MixedRealityToolkit Profile:

![image](https://user-images.githubusercontent.com/39840334/124646978-326ecb00-de4a-11eb-99c5-06f99c9af189.png)

Though you may hit some undefined behavior. We'll open an issue tracking and following up on the usability of near interactions with controllers!

As far as physics based collisions (OnCollisionEnter and OnTriggerEnter calls), you might be able to homebrew a solution by adding your own prefab with a collider onto the Oculus Quest prefab, though we don't officially support adding those via our API's at the moment.

**Comment by RogPodge** [comment]
Quick update on this, our near pointers currently don't track the controller positions well. New PR's will be coming through to address these issues and make near pointers usable with controllers 👍 👍 👍

**Comment by RogPodge** [comment]
PR out: https://github.com/microsoft/MixedRealityToolkit-Unity/pull/10059

**Comment by Efr3D** [comment]
Hi @RogPodge ,

I come back on this subject.

I'm not sure to understand how you configure your pointers.
I have saw your video on #10059 , But I can't reproduce it.
When I approach objects in my scene with controllers it does not detect that I am near to the object, whereas in Hand Tracking it is.

Have you any idea why I haven't the same behaviour when I use the Hand tracking and the controller ?

# Video
https://user-images.githubusercontent.com/10689982/150575897-d133032f-5beb-49a9-b437-38108d032464.mp4



# Tools
- Unity 2019.4.34f1
- Oculus QUEST 2 V35
-  SDK Oculus V35
- MRTK 2.7.3

# Settings

![image](https://user-images.githubusercontent.com/10689982/150575565-27aba517-c99a-443d-9977-4e49bd4c6558.png)

![image](https://user-images.githubusercontent.com/10689982/150575672-bd7bd24b-b36d-4e59-89c3-fafcbcb254f4.png)


![image](https://user-images.githubusercontent.com/10689982/150575695-902298ed-f9de-4c86-a10d-a4c8fcff9b7c.png)

![image](https://user-images.githubusercontent.com/10689982/150575722-cf26e00d-20e8-4feb-8082-1847b820e6fd.png)

![image](https://user-images.githubusercontent.com/10689982/150575749-d5fb4c74-cf34-4b76-b640-540964e98809.png)

# code

on my sphere, I used OnTouchStarted / OnTouchCompleted
IMixedRealityTouchHandler

**Comment by RogPodge** [comment]
We're a bit occupied at the moment with MRTK v3 work. I have some time reserved over this next period to do some cleanup on MRTK v2 issues, so I'll revisit this soon.

**Reply by Efr3D** [reply]
Thank you for your return, so I put this topic on standby. 
Do not hesitate to come back to me, when you come back to the subject.
